test/python/circuit/test_parameters.py

```diff
@@ -185,6 +185,22 @@ def test_bind_parameters_anonymously(self):
                 bqc_list = getattr(qc, assign_fun)(param_dict)
                 self.assertEqual(bqc_anonymous, bqc_list)
 
+    def test_bind_half_single_precision(self):
+        """Test binding with 16bit and 32bit floats."""
+        phase = Parameter("phase")
+        x = Parameter("x")
+        y = Parameter("y")
+        z = Parameter("z")
+        v = ParameterVector("v", 3)
+        for i in (numpy.float16, numpy.float32):
+            with self.subTest(float_type=i):
+                expr = (v[0] * (x + y + z) + phase) - (v[2] * v[1])
+                params = numpy.array([0.1 * i for i in range(8)], dtype=i)
```

Comment:
```suggestion
                params = numpy.array([0.1 * j for j in range(8)], dtype=i)
```
Not a bug, but might be less confusing to not re-use `i` here.